aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java

```diff
@@ -276,16 +276,20 @@ private SelectionSet wrapPagination(SelectionSet node) {
          */
         private static boolean isCustomType(@NonNull Field field) {
             Class<?> cls = getClassForField(field);
-            if (Model.class.isAssignableFrom(cls) || Enum.class.isAssignableFrom(cls)) {
+            if (cls.getClassLoader() == null || isDateType(cls)) {
                 return false;
             }
-            try {
-                JavaFieldType.from(cls.getSimpleName());
+            if (Model.class.isAssignableFrom(cls) || Enum.class.isAssignableFrom(cls)) {
                 return false;
-            } catch (IllegalArgumentException exception) {
-                // if we get here then field is  a custom type
-                return true;
             }
+            return true;
+        }
+
+        private static boolean isDateType(Class<?> cls) {
+            return cls.getSimpleName().equals(JavaFieldType.DATE.stringValue()) ||
```

Comment:
Maybe something like this to make it a little DRYer?
```
        Set<String> nonNativeTypeNames = new HashSet<>(Arrays.asList(
                DATE.stringValue(),
                DATE_TIME.stringValue(),
                TIME.stringValue(),
                TIMESTAMP.stringValue()));
        return nonNativeTypeNames.contains(clazz.getSimpleName());
```

Reply:
> Maybe something like this to make it a little DRYer?
> 
> ```
>         Set<String> nonNativeTypeNames = new HashSet<>(Arrays.asList(
>                 DATE.stringValue(),
>                 DATE_TIME.stringValue(),
>                 TIME.stringValue(),
>                 TIMESTAMP.stringValue()));
>         return nonNativeTypeNames.contains(clazz.getSimpleName());
> ```

sounds good, also i noticed that we are only comparing the class's name , shall we use package path + class name ? cause we may have classes that have duplicated names but different packages

Reply:
That is a valid point.  I'm not really sure why `JavaFieldType` has a `String javaFieldType` property instead of a `Class javaFieldClass` property.  Maybe you could try making that change?  Then we could just compare the `Class` directly, which would always be accurate.

Reply:
@richardmcclellan sure will add it in the next revision